src/dsp_tools/commands/xmlupload/models/serialise/serialise_resource.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from dsp_tools.models.datetimestamp import DateTimeStamp
+
+
+@dataclass(frozen=True)
+class SerialiseResource:
+    res_id: str
+    res_type: str
+    label: str
+    permissions: str | None
+    project_iri: str
+    migration_metadata: MigrationMetadata | None
+
+    def serialise(self) -> dict[str, Any]:
+        serialised = self.migration_metadata.to_dict() if self.migration_metadata else {}
+        serialised.update(self._get_resource_info())
+        return serialised
+
+    def _get_resource_info(self) -> dict[str, Any]:
+        serialised = {
+            "rdfs:label": self.label,
+            "@type": self.res_type,
+            "knora-api:attachedToProject": {"@id": self.project_iri},
+        }
+        if self.permissions:
+            serialised["knora-api:hasPermissions"] = self.permissions
+        return serialised
+
+
+@dataclass(frozen=True)
+class MigrationMetadata:
```

Comment:
A class with this name exists already. This class here is for serialisation, so I suggest to add the same prefix than the other classes in this file.

```suggestion
class SerialiseMigrationMetadata:
```